assets/js/base/components/shipping-rates-control/package.js

```diff
@@ -64,14 +61,38 @@ const Package = ( {
 					} ) }
 				</ul>
 			) }
-			<PackageOptions
-				className={ className }
-				noResultsMessage={ noResultsMessage }
-				onChange={ onChange }
-				options={ shippingRate.shipping_rates }
-				renderOption={ renderOption }
-				selected={ selected }
-			/>
+		</>
+	);
+	const body = (
+		<PackageOptions
+			className={ className }
+			noResultsMessage={ noResultsMessage }
+			onChange={ onChange }
+			options={ shippingRate.shipping_rates }
+			renderOption={ renderOption }
+			selected={ selected }
+		/>
+	);
+	if ( collapsible ) {
+		return (
+			<PanelBody
+				className="wc-block-shipping-rates-control__package"
+				title={ header }
+				initialOpen={ true }
+			>
+				<PanelRow>{ body }</PanelRow>
+			</PanelBody>
+		);
+	}
+	return (
+		<div
+			className={ classNames(
+				'wc-block-shipping-rates-control__package',
+				className
+			) }
+		>
+			{ header }
+			{ body }
 		</div>
```

Comment:
`PanelBody` has an [opened](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/panel#opened) prop that forces the panel to remain open, would that be a better option here instead of this conditional?

Reply:
I was experimenting with this idea but I'm not sure that would help simplify the code. The thing is that `<PanelBody>` and `<PanelRow>` have many styles applied that would need to be reset in the _Checkout_ block: background colors, margins & paddings, hiding the arrow icon, etc.

In addition to that, `<PanelBody>` keeps wrapping the header in a `<button>` even when the `opened` prop is set to true, that would affect accessibility negatively, because when rates are not collapsible, the header should just be text, not a `<button>`.

What do you think about leaving it as-is?

Reply:
If you've already tried and it doesn't work well, that's good enough for me 👍